Source/Ulibs.Tests/Ulibs.Tests.csproj

```diff
@@ -1,83 +1,24 @@
-    <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-  </PropertyGroup>
-  <ItemGroup>
-    <Reference Include="Microsoft.CSharp" />
-    <Reference Include="Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
-      <HintPath>..\packages\Newtonsoft.Json.11.0.1\lib\net45\Newtonsoft.Json.dll</HintPath>
-    </Reference>
-    <Reference Include="nunit.framework, Version=3.9.0.0, Culture=neutral, PublicKeyToken=2638cd05610744eb, processorArchitecture=MSIL">
-      <HintPath>..\packages\NUnit.3.9.0\lib\net45\nunit.framework.dll</HintPath>
-    </Reference>
-    <Reference Include="System" />
-    <Reference Include="System.Core" />
-    <Reference Include="System.Xml.Linq" />
-    <Reference Include="System.Data" />
-    <Reference Include="System.Xml" />
-  </ItemGroup>
-  <ItemGroup>
-    <Compile Include="FullExceptionString\ExtensionsTests.cs" />
-    <Compile Include="Properties\AssemblyInfo.cs" />
-    <Compile Include="ShellEscape\StringExtensionsTests.cs" />
-    <Compile Include="TinyJsonDeser\JsonDeserializerTests.cs" />
-    <Compile Include="TinyJsonSer\CustomDictionary.cs" />
-    <Compile Include="TinyJsonSer\CustomEnumerable.cs" />
-    <Compile Include="TinyJsonSer\CustomReadOnlyCollection.cs" />
-    <Compile Include="TinyJsonSer\CustomReadOnlyDictionary.cs" />
-    <Compile Include="TinyJsonSer\CustomReadOnlyList.cs" />
-    <Compile Include="TinyJsonSer\JsonSerializerTests.cs" />
-  </ItemGroup>
-  <ItemGroup>
-    <None Include="packages.config" />
-  </ItemGroup>
+
   <ItemGroup>
-    <ProjectReference Include="..\ULibs.FullExceptionString\ULibs.FullExceptionString.csproj">
-      <Project>{40ea8064-a654-40d3-9849-9a0e52248de9}</Project>
-      <Name>ULibs.FullExceptionString</Name>
-    </ProjectReference>
-    <ProjectReference Include="..\ULibs.ShellEscape\ULibs.ShellEscape.csproj">
-      <Project>{fc0fa749-3a05-4a1d-bd43-f73473936dbe}</Project>
-      <Name>ULibs.ShellEscape</Name>
-    </ProjectReference>
-    <ProjectReference Include="..\ULibs.TinyJsonDeser\ULibs.TinyJsonDeser.csproj">
-      <Project>{f46791e7-5434-400b-b5b4-6659909c262e}</Project>
-      <Name>ULibs.TinyJsonDeser</Name>
-    </ProjectReference>
-    <ProjectReference Include="..\ULibs.TinyJsonSer\ULibs.TinyJsonSer.csproj">
-      <Project>{3960af49-22de-489e-8348-35e32c605b4c}</Project>
-      <Name>ULibs.TinyJsonSer</Name>
-    </ProjectReference>
+    <PackageReference Include="Microsoft.CSharp" Version="4.7.0" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.5.0" />
+    <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
+    <PackageReference Include="NUnit" Version="3.12.0" />
+    <PackageReference Include="NUnit3TestAdapter" Version="3.16.1" />
+    <PackageReference Include="TeamCity.VSTest.TestAdapter" Version="1.0.20" />

```

Comment:
Ooh, does this mean that test results get auto-imported into teamcity? Might have to look into using that in our build scripts...

Reply:
Yep indeedy. On local runs, it does nothing. But when it detects a TC environment variable it emits a detailed set of service messages that TC pick up and gives you live test progress as they run. :)